pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -107,6 +107,7 @@ stdenv.mkDerivation ({
     ${optionalString (hasActiveLibrary && hyperlinkSource) "export PATH=${hscolour}/bin:$PATH"}
 
     configureFlags="--verbose --prefix=$out --libdir=\$prefix/lib/\$compiler --libsubdir=\$pkgid $configureFlags"
+    configureFlags+=" --with-gcc=$CC"
```

Comment:
Cabal does not find out about the proper compiler without that setting? Can you do me a favor and paste a package build log without that setting? I'd be curious to know what the error messages look like.